tests/test_util.py

```diff
@@ -0,0 +1,180 @@
+    assert result.dtype == np.int8
+    assert_array_equal(result, inp)
+
+    dtype = np.dtype([("col1", np.int16), ("col2", np.float32)])
+    result = util.gentle_asarray(inp, dtype=dtype)
+    assert result.dtype == dtype
+    assert result["col1"].dtype == np.int16
+    assert_array_equal(result["col1"], np.array([1, 2, 3, 4], dtype=np.int16))
+    assert result["col2"].dtype == np.float32
+    assert_array_equal(result["col1"], np.array([1, 2, 3, 4], dtype=np.float32))
+
+
+def test_gentle_asarray_recarray_input():
+    dtype = np.dtype([("col1", np.int16), ("col2", np.float32)])
+    inp = np.array([1, 2, 3, 4], dtype=dtype)
+    result = util.gentle_asarray(inp, dtype=dtype)
+    assert result.dtype == dtype
+    assert_array_equal(result, inp)
+
+
+def test_gentle_asarray_fits_rec_input():
+    cols = [fits.Column("col1", format="e", array=np.array([1, 2, 3, 4]))]
+    inp = fits.FITS_rec.from_columns(cols)
+    out_dtype = np.dtype([("col1", np.float32)])
+    result = util.gentle_asarray(inp, dtype=out_dtype)
+    assert result.dtype == out_dtype
+    assert_array_equal(result, inp)
+
+
+def test_gentle_asarray_fits_rec_pseudo_unsigned():
+    cols = [fits.Column("col1", format="j", array=np.array([1, 2, 3, 4], np.uint32), bzero=2147483648)]
+    inp = fits.FITS_rec.from_columns(cols)
+    out_dtype = np.dtype([("col1", np.uint32)])
+    result = util.gentle_asarray(inp, dtype=out_dtype)
+    assert result["col1"].dtype == np.uint32
+
+
+def test_gentle_asarray_nested_array():
+    in_dtype = np.dtype([("col1", np.dtype("2f")), ("col2", np.int16)])
+    inp = np.array([1, 2, 3, 4], dtype=in_dtype)
+    out_dtype = np.dtype([("col1", np.dtype("2f")), ("col2", np.int8)])
+    result = util.gentle_asarray(inp, dtype=out_dtype)
+    assert result.dtype == out_dtype
+    assert_array_equal(result["col2"], inp["col2"])
+
+
+def test_gentle_asarray_mismatched_column_names():
+    in_dtype = np.dtype([("col1", np.int16), ("col2", np.float32)])
+    inp = np.array([1, 2, 3, 4], dtype=in_dtype)
+    out_dtype = np.dtype([("col1", np.int16), ("foo", np.float32)])
+    with pytest.raises(ValueError):
+        util.gentle_asarray(inp, dtype=out_dtype)
+
+
+def test_gentle_asarray_column_name_case():
+    in_dtype = np.dtype([("col1", np.int16), ("col2", np.float32)])
+    inp = np.array([1, 2, 3, 4], dtype=in_dtype)
+    out_dtype = np.dtype([("COL1", np.int16), ("COL2", np.float32)])
+    result = util.gentle_asarray(inp, dtype=out_dtype)
+    assert result.dtype == out_dtype
+    assert_array_equal(result, inp)
+
+
+def test_gentle_asarray_scalar_input():
+    result = util.gentle_asarray(3.14159, dtype=np.float32)
+    assert result.dtype == np.float32
+    assert_array_equal(result, np.array([3.14159], dtype=np.float32))
+
+
+def test_gentle_asarray_invalid_conversion():
+    with pytest.raises(ValueError):
+        util.gentle_asarray(object(), dtype=np.float32)
+
+
+def test_get_short_doc():
```

Comment:
Is there a reason to have `get_short_doc` is not over in the `fits_support.py` module?  It's only used for populating the FITS keyword comment when writing out files to FITS format I believe.

Reply:
It does appear to only be used in fits_support.py, so I can quarantine it over there.